web/packages/shared/components/UnifiedResources/ResourceCard.story.tsx

```diff
@@ -86,35 +92,51 @@ const Grid = styled.div`
 export default meta;
 type Story = StoryObj<typeof ResourceCard>;
 
+const ActionButton = <ButtonBorder size="small">Action</ButtonBorder>;
+
 export const Cards: Story = {
   render() {
-    const ctx = createTeleportContext();
     return (
-      <MemoryRouter>
-        <TeleportContextProvider ctx={ctx}>
-          <Grid gap={2}>
-            {[
-              ...apps,
-              ...databases,
-              ...kubes,
-              ...nodes,
-              ...additionalResources,
-              ...desktops,
-            ].map((res, i) => (
-              <ResourceCard
-                key={i}
-                resource={res}
-                pinned={false}
-                pinResource={() => {}}
-                pinningDisabled={false}
-                selectResource={() => {}}
-                selected={false}
-                pinningNotSupported={false}
-              />
-            ))}
-          </Grid>
-        </TeleportContextProvider>
-      </MemoryRouter>
+      <Grid gap={2}>
+        {[
+          ...apps.map(resource =>
+            makeUnifiedResourceCardApp(resource, { ActionButton })
+          ),
+          ...databases.map(resource =>
+            makeUnifiedResourceCardDatabase(resource, {
+              ActionButton,
+            })
+          ),
+          ...kubes.map(resource =>
+            makeUnifiedResourceCardKube(resource, { ActionButton })
+          ),
+          ...nodes.map(resource =>
+            makeUnifiedResourceCardNode(resource, { ActionButton })
+          ),
+          ...additionalResources.map(resource =>
+            makeUnifiedResourceCardApp(resource, { ActionButton })
+          ),
+          ...desktops.map(resource =>
+            makeUnifiedResourceCardDesktop(resource, { ActionButton })
+          ),
+        ].map((res, i) => (
+          <ResourceCard
+            key={i}
+            pinned={false}
+            pinResource={() => {}}
+            pinningDisabled={false}
+            selectResource={() => {}}
+            selected={false}
+            pinningNotSupported={false}
```

Comment:
We will need a third `pinning` related option along the lines of `pinningHidden` or something like that. Might need to rethink the phrase. The idea is that we won't be able to pin (or have the pinned tab) in access requests. Right now, the access request component looks a lot like this story component where we just pass in empty/false. I'm not sure it needs it's own level of abstraction but something to chew on

Reply:
I think we should convert these `pinning*` props to enum:
```ts
export enum PinningSupport {
  Supported = 'Supported',
  /**
   * Disables pinning functionality if a leaf cluster hasn't been upgraded yet.
   * Shows an appropriate message on hover.
   * */
  NotSupported = 'NotSupported',
  /** Disables the pinning button. */
  Disabled = 'Disabled',
  /** Hides the pinning button completely. */
  Hidden = 'Hidden',
}
```

I also added support for hiding.